src/test/librbd/test_mock_ManagedLock.cc

```diff
@@ -393,6 +393,27 @@ TEST_F(TestMockManagedLock, AcquireLockBlocklist) {
   ASSERT_EQ(0, when_shut_down(managed_lock));
 }
 
+TEST_F(TestMockManagedLock, SendAcquireLockBlocklist) {
+  librbd::ImageCtx *ictx;
+  ASSERT_EQ(0, open_image(m_image_name, &ictx));
+
+  MockManagedLockImageCtx mock_image_ctx(*ictx);
+  MockMockManagedLock managed_lock(ictx->md_ctx, *ictx->asio_engine,
+                               ictx->header_oid, mock_image_ctx.image_watcher,
+                               librbd::managed_lock::EXCLUSIVE, true, 0);
+
+  InSequence seq;
+
+  expect_get_watch_handle(*mock_image_ctx.image_watcher, 0);
+  C_SaferCond acquire_ctx;
+
+  EXPECT_CALL(*mock_image_ctx.image_watcher, is_blocklisted()).WillOnce(Return(true));
+  managed_lock.acquire_lock(&acquire_ctx);
+  acquire_ctx.wait();
```

Comment:
Use `when_acquire_lock` helper instead of defining `acquire_ctx` and assert on the returned error:
```
  ASSERT_EQ(-EBLOCKLISTED, when_acquire_lock(managed_lock));
  ASSERT_FALSE(is_lock_owner(managed_lock));
```

Reply:
Done